miraclelinux/meta-debian
# openssh: Update version from 7.9p1-10+deb10u2 to 7.9p1-10+deb10u3

# Purpose of pull request
This version fixes following CVEs.
- CVE-2023-38408

DLA: DLA 3532-1

And, fix the recipe to ptest runs finish:
- Add patch to fix "cert not yet valid" test 
- Backport a patch for adds coreutils to RDEPENDS of ptest
- Add run-ptest file from OE-Core

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemuarm64 image
3. Run ptest and check ptest log

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
DISTRO_FEATURES_append = " pam"
IMAGE_INSTALL_append = " openssh-keygen openssh-misc openssh-scp openssh-sftp openssh-sftp-server openssh-ssh openssh-sshd"
IMAGE_INSTALL_append = " openssh-ptest"
```
```
$ bitbake core-image-minimal
```

### Run qemuarm64 image
```
$ runqemu qemuarm64 nographic
```

### Run ptest and check ptest log
The command to display ptest results was created with reference to meta-debian's tests/run_ptest.sh.
```
# test_package=openssh
# ptest-runner $test_package 2>&1 | tee ptest.log
# echo "$test_package: PASS/SKIP/FAIL = `grep '^PASS:' ptest.log | wc -l`/`grep '^SKIP:' ptest.log | wc -l`/`grep '^FAIL:' ptest.log | wc -l`"
```

## Test result
### Build package
Build succeeded.
```
$ bitbake openssh
Loading cache: 100% |############################################################################################################################################################################################################################################| Time: 0:00:00
Loaded 2381 entries from dependency cache.
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |##########################################################################################################################################################################################################################################| Time: 0:00:00
Parsing of 1361 .bb files complete (1355 cached, 6 parsed). 2381 targets, 72 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-openssh_7.9p1-10+deb10u3:efebf299bf9f3735d9eb0b85744a196af5a3e778"
meta-debian-extended = "HEAD:1faec1396bc9871f46202287446757993c392468"
meta-emlinux         = "HEAD:faf103e572ae8697df8bac3861f27555b90ed5de"
meta-emlinux-private = "HEAD:8207525289999d2c41983eb8155127b62621cdb1"

Initialising tasks: 100% |#######################################################################################################################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 6 Found 0 Missed 6 Current 447 (0% match, 98% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2074 tasks of which 2055 didn't need to be rerun and all succeeded.
```

### Package test
The ptest has passed (no FAILs).
```
root@qemuarm64:~# test_package=openssh
root@qemuarm64:~# ptest-runner $test_package 2>&1 | tee ptest.log
START: ptest-runner
2023-09-05T11:24
BEGIN: /usr/lib/openssh/ptest
...
PASS: authinfo
SKIP: agent-ptrace
DURATION: 4101
END: /usr/lib/openssh/ptest
2023-09-05T12:32
STOP: ptest-runner
root@qemuarm64:~# echo "$test_package: PASS/SKIP/FAIL = `grep '^PASS:' ptest.log | wc -l`/`grep '^SKIP:' ptest.log | wc -l`/`grep '^FAIL:' ptest.log | wc -l`"
openssh: PASS/SKIP/FAIL = 66/7/0
```
